tests/nnfw_api/lib/GenModelTrain.h

```diff
@@ -265,69 +368,73 @@ class GenModelTrain : public ::testing::Test
 
       const int num_epoch = _context->epoch();
       ASSERT_GE(num_epoch, 2);
-      const int num_step = num_expecteds / tri.batch_size;
-      ASSERT_GE(num_step, 1);
-
       // Set input values & expected output values, train, and check loss
       for (const auto &train_case : _context->train_cases())
       {
-        // Inputs
-        const auto &ref_inputs = train_case.inputs;
-        ASSERT_EQ(_so.inputs.size(), ref_inputs.size());
-        for (uint32_t i = 0; i < _so.inputs.size(); i++)
-        {
-          // Fill the values
-          ASSERT_EQ(_so.inputs[i].size(), ref_inputs[i].size());
-          memcpy(_so.inputs[i].data(), ref_inputs[i].data(), ref_inputs[i].size());
-        }
+        const auto &[inputs_dataset, outputs_dataset] = train_case.dataset;
+        const auto num_step = inputs_dataset.size();
+        ASSERT_EQ(num_step, outputs_dataset.size());
 
-        // Expected outputs
-        const auto &ref_expects = train_case.expects;
-        ASSERT_EQ(_so.expects.size(), ref_expects.size());
-        for (uint32_t i = 0; i < _so.expects.size(); i++)
-        {
-          // Fill the values
-          ASSERT_EQ(_so.expects[i].size(), ref_expects[i].size());
-          memcpy(_so.expects[i].data(), ref_expects[i].data(), ref_expects[i].size());
-        }
-
-        if (train_case.expected_fail_run())
-        {
-          ASSERT_NE(nnfw_train(_so.session, true), NNFW_STATUS_NO_ERROR);
-          continue;
-        }
-
-        // Train
+        // Prepare expected losses
         const auto &ref_losses = train_case.losses;
-        std::vector<float> actual_losses(num_expecteds, 0.f);
+        std::vector<float> actual_losses;
+
         for (uint32_t epoch = 0; epoch < num_epoch; ++epoch)
         {
-          NNFW_ENSURE_SUCCESS(nnfw_train(_so.session, true));
+          actual_losses.resize(num_expecteds, 0.f);
```

Comment:
There is no difference between the current size and given count, it does not initialize the value as `0.f`. It still contains the previous loss value, so now, the results between epochs are added together and printed out.

Here is the TensorFlow result with the same parameter:
```
Epoch 1/2 - loss: 403.3333
Epoch 2/2 - loss: 394.7772
```

Your loss `798.110534f` is the sum of the above two results.